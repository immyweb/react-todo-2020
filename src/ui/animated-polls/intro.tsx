import * as React from 'react';

import {
  APIntro,
  APHeadline,
  APCopy,
  APIntroCopy,
  APIntroImg,
  APIntroLink,
  APIntroHashTag,
} from './styles';

import { IIntroScreen } from './types';

export const IntroScreen: React.SFC<IIntroScreen> = ({
  imageMobile,
  imageDesktop,
  title,
  introText,
  cta,
  ctaLink,
  ctaUrl,
}) => {
  return (
    <APIntro>
      <APIntroImg imageMobile={imageMobile} imageDesktop={imageDesktop} />
      <APIntroCopy>
        <APHeadline>{title}</APHeadline>
        {introText.map((text, i) => (
          <APCopy key={i}>{text}</APCopy>
        ))}
        <APCopy>
          <APIntroLink href={ctaUrl}>
            {cta} <APIntroHashTag>{ctaLink}</APIntroHashTag>
          </APIntroLink>
        </APCopy>
      </APIntroCopy>
    </APIntro>
  );
};
