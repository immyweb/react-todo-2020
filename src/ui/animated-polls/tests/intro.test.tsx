import * as React from 'react';
import { shallow } from 'enzyme';

import { IntroScreen } from '../intro';

const introScreen = {
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png',
  imageDesktop: '',
  title: 'Jeremy Styles',
  introText: [
    'Jeremy Kyle needs help picking his moustache style.',
    'Flick through the gallery of images and vote on your favourite at the end.',
  ],
  cta: 'Read more from:',
  ctaLink: '#MOVEMBER',
  ctaUrl: 'http://www.thesun.co.uk',
};

function shallowRender(props?: any) {
  return shallow(<IntroScreen {...introScreen} {...props} />);
}

describe('<IntroScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });
});
