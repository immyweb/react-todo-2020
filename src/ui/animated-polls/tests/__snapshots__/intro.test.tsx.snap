// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    cta="Read more from:"
    ctaLink="#MOVEMBER"
    ctaUrl="http://www.thesun.co.uk"
    imageDesktop=""
    imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
    introText={
      Array [
        "Jeremy Kyle needs help picking his moustache style.",
        "Flick through the gallery of images and vote on your favourite at the end.",
      ]
    }
    title="Jeremy Styles"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          imageDesktop=""
          imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
        />,
        <glamorous(div)>
          <glamorous(glamorous(h2))>
            Jeremy Styles
          </glamorous(glamorous(h2))>
          <glamorous(glamorous(glamorous.P))>
            Jeremy Kyle needs help picking his moustache style.
          </glamorous(glamorous(glamorous.P))>
          <glamorous(glamorous(glamorous.P))>
            Flick through the gallery of images and vote on your favourite at the end.
          </glamorous(glamorous(glamorous.P))>
          <glamorous(glamorous(glamorous.P))>
            <glamorous(a)
              href="http://www.thesun.co.uk"
            >
              Read more from:
               
              <glamorous(span)>
                #MOVEMBER
              </glamorous(span)>
            </glamorous(a)>
          </glamorous(glamorous(glamorous.P))>
        </glamorous(div)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "imageDesktop": "",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(h2))>
              Jeremy Styles
            </glamorous(glamorous(h2))>,
            Array [
              <glamorous(glamorous(glamorous.P))>
                Jeremy Kyle needs help picking his moustache style.
              </glamorous(glamorous(glamorous.P))>,
              <glamorous(glamorous(glamorous.P))>
                Flick through the gallery of images and vote on your favourite at the end.
              </glamorous(glamorous(glamorous.P))>,
            ],
            <glamorous(glamorous(glamorous.P))>
              <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>
            </glamorous(glamorous(glamorous.P))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Jeremy Styles",
            },
            "ref": null,
            "rendered": "Jeremy Styles",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": "Jeremy Kyle needs help picking his moustache style.",
            },
            "ref": null,
            "rendered": "Jeremy Kyle needs help picking his moustache style.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": "Flick through the gallery of images and vote on your favourite at the end.",
            },
            "ref": null,
            "rendered": "Flick through the gallery of images and vote on your favourite at the end.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Read more from:",
                  " ",
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>,
                ],
                "href": "http://www.thesun.co.uk",
              },
              "ref": null,
              "rendered": Array [
                "Read more from:",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "#MOVEMBER",
                  },
                  "ref": null,
                  "rendered": "#MOVEMBER",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            imageDesktop=""
            imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png"
          />,
          <glamorous(div)>
            <glamorous(glamorous(h2))>
              Jeremy Styles
            </glamorous(glamorous(h2))>
            <glamorous(glamorous(glamorous.P))>
              Jeremy Kyle needs help picking his moustache style.
            </glamorous(glamorous(glamorous.P))>
            <glamorous(glamorous(glamorous.P))>
              Flick through the gallery of images and vote on your favourite at the end.
            </glamorous(glamorous(glamorous.P))>
            <glamorous(glamorous(glamorous.P))>
              <glamorous(a)
                href="http://www.thesun.co.uk"
              >
                Read more from:
                 
                <glamorous(span)>
                  #MOVEMBER
                </glamorous(span)>
              </glamorous(a)>
            </glamorous(glamorous(glamorous.P))>
          </glamorous(div)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "imageDesktop": "",
            "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/10/IntroScreen@2.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(glamorous(h2))>
                Jeremy Styles
              </glamorous(glamorous(h2))>,
              Array [
                <glamorous(glamorous(glamorous.P))>
                  Jeremy Kyle needs help picking his moustache style.
                </glamorous(glamorous(glamorous.P))>,
                <glamorous(glamorous(glamorous.P))>
                  Flick through the gallery of images and vote on your favourite at the end.
                </glamorous(glamorous(glamorous.P))>,
              ],
              <glamorous(glamorous(glamorous.P))>
                <glamorous(a)
                  href="http://www.thesun.co.uk"
                >
                  Read more from:
                   
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>
                </glamorous(a)>
              </glamorous(glamorous(glamorous.P))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Jeremy Styles",
              },
              "ref": null,
              "rendered": "Jeremy Styles",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": "Jeremy Kyle needs help picking his moustache style.",
              },
              "ref": null,
              "rendered": "Jeremy Kyle needs help picking his moustache style.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": "Flick through the gallery of images and vote on your favourite at the end.",
              },
              "ref": null,
              "rendered": "Flick through the gallery of images and vote on your favourite at the end.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <glamorous(a)
                  href="http://www.thesun.co.uk"
                >
                  Read more from:
                   
                  <glamorous(span)>
                    #MOVEMBER
                  </glamorous(span)>
                </glamorous(a)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Read more from:",
                    " ",
                    <glamorous(span)>
                      #MOVEMBER
                    </glamorous(span)>,
                  ],
                  "href": "http://www.thesun.co.uk",
                },
                "ref": null,
                "rendered": Array [
                  "Read more from:",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "#MOVEMBER",
                    },
                    "ref": null,
                    "rendered": "#MOVEMBER",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
